plonky2x/src/backend/function/mod.rs

```diff
@@ -174,39 +176,33 @@ impl<C: Circuit> VerifiableFunction<C> {
             AlgebraicHasher<InnerParameters::Field>,
         OuterParameters::Config: Serialize,
     {
-        let path = format!("{}/main.circuit", args.build_dir);
```

Comment:
nit:  Seems like there is alot of duplicated code between `prove` and `prove_wrapped`.  Might want to consider consolidating them a bit, if possible.